src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -1,18 +1,60 @@
 package wolox.training.controllers;
 
-import org.springframework.stereotype.Controller;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.*;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
 
 
-@Controller
+@RestController
+@RequestMapping("/api/books")
 public class BookController {
+    @Autowired
+    private BookRepository bookRepository;
 
-  @GetMapping("/greeting")
-  public String greeting(@RequestParam(defaultValue="Wolox") String name, Model model) {
-    model.addAttribute("name", name);
-    // The name "greeting" would not work
-    return "hello";
-  }
+    @GetMapping("/greeting")
+    public String greeting(@RequestParam(defaultValue = "Wolox") String name, Model model) {
+        model.addAttribute("name", name);
+        // The name "greeting" would not work
+        return "hello";
+    }
+
+
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable Long id) throws BookNotFoundException {
+        return bookRepository.findById(id)
+                .orElseThrow(BookNotFoundException::new);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return bookRepository.save(book);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        bookRepository.findById(id)
+                .orElseThrow(BookNotFoundException::new);
+        bookRepository.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public Book updateBook(@org.jetbrains.annotations.NotNull @RequestBody Book book, @PathVariable Long id) {
```

Comment:
Again, not sure about this annotation. Surely there's a different one?

Reply:
Ohhh, this is the one! Nope, that's not the one we use, AFAIK.